syntax = "proto3";

package message;

service Message {
    // Отправка файла на сервер и получение подтверждения о получении
    rpc UploadFile (stream FileUploadRequest) returns (FileSuccessReply);
}

enum Status {
    Unknown = 0;
    Ok = 1;
    Failed = 2;
}

message MetaData {
    string file_name = 1;
    string md5 = 2;
    bool backup = 3;
}

message FileUploadRequest {
    oneof request {
        MetaData metadata = 1;
        bytes chunk = 2;
    }
}

message FileSuccessReply {
    Status status = 1;
}
